src/main/java/formflow/library/ScreenController.java

```diff
@@ -714,41 +1143,214 @@ ModelAndView navigation(
+            Collections.EMPTY_LIST);
+        if (!repeatForIterationsData.isEmpty()) {
+          Map<String, Object> nextRepeatForIterationScreen = subflowManager.getNextRepeatForIterationUuid(
+              repeatForSaveDataAs,
+              currentSubflowEntryData);
+          if (nextRepeatForIterationScreen != null) {
+            return String.format("/flow/%s/%s/%s/%s", flowName, nextScreenConfiguration.getName(),
+                subflowIterationUuid,
+                nextRepeatForIterationScreen.get("uuid"));
+          } else {
+            return String.format("/flow/%s/%s", flowName,
+                subflowManager.getSubflowConfiguration(flowName, currentSubflowName).getReviewScreen());
+          }
+        } else {
+          return String.format("/flow/%s/%s/%s", flowName, nextScreenConfiguration.getName(),
+              subflowIterationUuid);
+        }
+      }
+
+      String validatedRepeatForIterationUuid = getValidatedRepeatForIterationUuid(submission, flowName, currentSubflowName,
+          subflowIterationUuid, repeatForSaveDataAs, repeatForIterationUuid);
+
+      if (validatedRepeatForIterationUuid == null) {
+        throwNotFoundError(submission.getFlow(), currentScreenConfiguration.getName(),
+            String.format(
+                "RepeatFor iteration with uuid ('%s') not found under subflow '%s' iteration uuid ('%s') in flow '%s', when navigating to '%s'",
+                repeatForIterationUuid, currentScreenConfiguration.getSubflow(), subflowIterationUuid, submission.getFlow(),
+                currentScreenConfiguration.getName()));
+      }
+
+      Boolean inLastScreenOfRepeatsForIteration = nextScreenConfiguration.getSubflow() == null;
+      if (inLastScreenOfRepeatsForIteration) {
+        Map<String, Object> repeatForIteration = subflowManager.getRepeatForIteration(currentSubflowEntryData,
+            repeatForSaveDataAs, validatedRepeatForIterationUuid);
+        repeatForIteration.put(ITERATION_IS_COMPLETE_KEY, true);
         submission = saveToRepository(submission);
-        redirectString = String.format("/flow/%s/%s", flow, nextScreen);
-        if (subflowManager.subflowHasRelationship(flow, currentSubflowName)) {
-          if (!subflowManager.hasFinishedAllSubflowIterations(currentSubflowName, submission)) {
-            // If you are in a subflow with a relationship we want you to keep looping until you loop over every iteration in the related subflow
-            redirectString = String.format("/flow/%s/%s", flow, subflowManager.getIterationStartScreenForSubflow(flow, currentSubflowName));
+
+        boolean allRepeatForIterationsComplete = subflowManager.hasFinishedAllIterations(repeatForSaveDataAs,
+            currentSubflowEntryData);
+
+        if (allRepeatForIterationsComplete) {
+          submission.setIterationIsCompleteToTrue(currentSubflowName, subflowIterationUuid);
+          saveToRepository(submission);
+
+          if (subflowRelationship.getRelatesTo() != null) {
```

Comment:
Double check this one since we dont expect to have repeats for without relates to